spark/dl/src/main/scala/com/intel/analytics/bigdl/utils/serializer/DataConverter.scala

```diff
@@ -0,0 +1,620 @@
+      val regularizerType = regularizer.getRegularizerType
+      if (regularizer.getRegularDataCount == 0) {
+        return null
+      }
+      val l1 = regularizer.getRegularDataList.get(0)
+      val l2 = regularizer.getRegularDataList.get(1)
+      regularizerType match {
+        case serialization.Model.RegularizerType.L1Regularizer => L1Regularizer[T](l1)
+        case serialization.Model.RegularizerType.L2Regularizer => L2Regularizer[T](l2)
+        case serialization.Model.RegularizerType.L1L2Regularizer => L1L2Regularizer[T](l1, l2)
+      }
+    }
+
+    override def setAttributeValue[T : ClassTag]
+    (attributeBuilder: AttrValue.Builder, value: Any,
+     valueType : universe.Type = null)
+    (implicit ev: TensorNumeric[T]): Unit = {
+      attributeBuilder.setDataType(DataType.REGULARIZER)
+      if (value != null) {
+        var regularizerBuilder = serialization.Model.Regularizer.newBuilder
+        val regularizer = value.asInstanceOf[L1L2Regularizer[T]]
+        val l1 = regularizer.l1
+        val l2 = regularizer.l2
+        regularizerBuilder.addRegularData(l1)
+        regularizerBuilder.addRegularData(l2)
+        val regularizerType = regularizer match {
+          case l1: L1Regularizer[_] => serialization.Model.RegularizerType.L1Regularizer
+          case l2: L2Regularizer[_] => serialization.Model.RegularizerType.L2Regularizer
+          case l1l2: L1L2Regularizer[_] => serialization.Model.RegularizerType.L1L2Regularizer
+        }
+        regularizerBuilder.setRegularizerType(regularizerType)
+        attributeBuilder.setRegularizerValue(regularizerBuilder.build)
+      }
+    }
+
+  }
+
+/**
+ * DataConverter for [[com.intel.analytics.bigdl.tensor.Tensor]]
+ */
+  object TensorConverter extends DataConverter {
+
+    override def getAttributeValue[T: ClassTag](attribute: AttrValue)
+      (implicit ev: TensorNumeric[T]): AnyRef = {
+      val serializedTensor = attribute.getTensorValue
+      val data = serializedTensor.getDataList.asScala
+      if (data.size == 0) {
+        return null
+      }
+      val sizes = serializedTensor.getSizeList.asScala
+      val strorageArray = new Array[T](data.size)
+      var i = 0;
+      while (i < data.size) {
+        strorageArray(i) = ev.fromType[Double](data(i))
+        i += 1
+      }
+      val sizeArray = new Array[Int](sizes.size)
+      i = 0;
+      while (i < sizes.size) {
+        sizeArray(i) = sizes(i)
+        i += 1
+      }
+      Tensor[T](strorageArray, sizeArray)
+    }
+
+    override def setAttributeValue[T: ClassTag]
+      (attributeBuilder: AttrValue.Builder, value: Any,
+       valueType : universe.Type = null)
+      (implicit ev: TensorNumeric[T]): Unit = {
+      attributeBuilder.setDataType(DataType.TENSOR)
+      if (value != null) {
+        val tensor = value.asInstanceOf[Tensor[T]]
+        val tensorBuilder = BigDLTensor.newBuilder
+        tensor.storage().array().foreach(data => tensorBuilder.addData(ev.toType[Double](data)))
```

Comment:
I'm afraid the performance. Should we use byte array to store the tensor conten, like tensorflow